eva/optimizer/rules/rules_manager.py

```diff
@@ -77,6 +78,7 @@ class RulesManager:
     def __init__(self):
         self._logical_rules = [
             LogicalInnerJoinCommutativity(),
+            CacheFunctionExpressionInApply(),
```

Comment:
Make sure I understand correctly. Since the cache cost is 0, for now this works like a rewrite rule.

Reply:
Yes, but I feel we might have some logic based on the cost in future. So wanted to keep it as a logical expression. Sounds good?